src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -91,7 +91,9 @@ char* CDSConfig::default_archive_path() {
     char *end = strrchr(jvm_path, *os::file_separator());
     if (end != nullptr) *end = '\0';
     stringStream tmp;
-    tmp.print("%s%sclasses", jvm_path, os::file_separator());
+    const char* subdir = WINDOWS_ONLY("bin") NOT_WINDOWS("lib");
+    tmp.print("%s%s%s%s%s%sclasses", Arguments::get_java_home(), os::file_separator(), subdir,
+              os::file_separator(), JVM_VARIANT, os::file_separator());
```

Comment:
I think the above call to os::jvm_path (lines 90-92) can also be removed.

Reply:
Also, the definitions of JVM_VARIANT in os_bsd.cpp should be removed.

Reply:
I've fixed both.